pandas/core/strings.py

```diff
@@ -2107,7 +2107,9 @@ def _get_series_list(self, others):
         elif isinstance(others, np.ndarray) and others.ndim == 2:
             others = DataFrame(others, index=idx)
             return [others[x] for x in others]
-        elif is_list_like(others, allow_sets=False):
+        elif is_list_like(others, allow_sets=False) and not isinstance(
```

Comment:
Not sure about this. I *think* the real issue is that `is_list_like` returns True for `s.str` when it shouldn't, though I do see that is also technically Iterable so maybe this should be allowed and just fixed